src/Workspaces/Core/Portable/Workspace/Solution/ChecksumCollection.cs

```diff
@@ -82,47 +82,4 @@ IEnumerator IEnumerable.GetEnumerator()
             }
         }
     }
-
-    // we have a type for each kind so that we can distinguish these later
-    internal class ProjectChecksumCollection : ChecksumCollection
-    {
-        public ProjectChecksumCollection(Checksum[] checksums) : this((object[])checksums) { }
-        public ProjectChecksumCollection(object[] checksums) : base(WellKnownSynchronizationKind.ProjectChecksumCollection, checksums) { }
-    }
-
-    internal class DocumentChecksumCollection : ChecksumCollection
-    {
-        public DocumentChecksumCollection(Checksum[] checksums) : this((object[])checksums) { }
-        public DocumentChecksumCollection(object[] checksums) : base(WellKnownSynchronizationKind.DocumentChecksumCollection, checksums) { }
-    }
-
-    internal class TextDocumentChecksumCollection : ChecksumCollection
-    {
-        public TextDocumentChecksumCollection(Checksum[] checksums) : this((object[])checksums) { }
-        public TextDocumentChecksumCollection(object[] checksums) : base(WellKnownSynchronizationKind.TextDocumentChecksumCollection, checksums) { }
-    }
-
-    internal class AnalyzerConfigDocumentChecksumCollection : ChecksumCollection
-    {
-        public AnalyzerConfigDocumentChecksumCollection(Checksum[] checksums) : this((object[])checksums) { }
-        public AnalyzerConfigDocumentChecksumCollection(object[] checksums) : base(WellKnownSynchronizationKind.AnalyzerConfigDocumentChecksumCollection, checksums) { }
-    }
-
-    internal class ProjectReferenceChecksumCollection : ChecksumCollection
-    {
-        public ProjectReferenceChecksumCollection(Checksum[] checksums) : this((object[])checksums) { }
-        public ProjectReferenceChecksumCollection(object[] checksums) : base(WellKnownSynchronizationKind.ProjectReferenceChecksumCollection, checksums) { }
-    }
-
-    internal class MetadataReferenceChecksumCollection : ChecksumCollection
-    {
-        public MetadataReferenceChecksumCollection(Checksum[] checksums) : this((object[])checksums) { }
-        public MetadataReferenceChecksumCollection(object[] checksums) : base(WellKnownSynchronizationKind.MetadataReferenceChecksumCollection, checksums) { }
-    }
-
-    internal class AnalyzerReferenceChecksumCollection : ChecksumCollection
-    {
-        public AnalyzerReferenceChecksumCollection(Checksum[] checksums) : this((object[])checksums) { }
-        public AnalyzerReferenceChecksumCollection(object[] checksums) : base(WellKnownSynchronizationKind.AnalyzerReferenceChecksumCollection, checksums) { }
-    }
```

Comment:
we could consider still having these, just for the slight amount of strong type safety it affords in one locatin.  but it seemed pretty overkill to me.